contrib/registry.py

```diff
@@ -76,3 +66,104 @@ def model(*models_factories: Factory) -> Union[Factory, List[Factory]]:
             :returns: single model factory or list of them
         """
         return Registry._inner_register("model", *models_factories)
+
+    @staticmethod
+    def module(*modules_factories: Factory) -> Union[Factory, List[Factory]]:
+        """Add module type or factory method to global
+            module list to make it available in config
+            Can be called or used as decorator
+            :param: modules_factories
+                Required module factory (method or type)
+            :returns: single module factory or list of them
+        """
+        return Registry._inner_register("module", *modules_factories)
+
+    @staticmethod
+    def optimizer(
+            *optimizer_factories: Factory
+    ) -> Union[Factory, List[Factory]]:
+        """Add optimizer type or factory method to global
+            optimizer list to make it available in config
+            Can be called or used as decorator
+            :param: optimizer_factories
+                Required optimizer factory (method or type)
+            :returns: single optimizer factory or list of them
+        """
+        return Registry._inner_register("optimizer", *optimizer_factories)
+
+    @staticmethod
+    def name2nn(name):
+        if name is None:
+            return None
+        elif isinstance(name, nn.Module):
+            return name
+        elif isinstance(name, str):
+            return _REGISTERS["module"][name]
+        else:
+            return name
+
+    @staticmethod
+    def create_callback(callback=None, **callback_params):
+        if callback is None:
+            return None
+
+        callback = _REGISTERS["callback"][callback](**callback_params)
+        return callback
+
+    @staticmethod
+    def create_criterion(criterion=None, **criterion_params):
+        if criterion is None:
+            return None
+
+        criterion = _REGISTERS["criterion"][criterion](**criterion_params)
+        if torch.cuda.is_available():
+            criterion = criterion.cuda()
+        return criterion
+
+    @staticmethod
+    def create_model(config, available_networks=None):
+        model_params = config.pop("model_params", {})
+        model_name = model_params.pop("model", None)
+        fp16 = model_params.pop("fp16", False) and torch.cuda.is_available()
+
+        available_networks = available_networks or {}
+        available_networks = {**available_networks, **_REGISTERS["model"]}
+
+        model = available_networks[model_name](**model_params)
+
+        if fp16:
+            model = Fp16Wrap(model)
+
+        return model
+
+    @staticmethod
+    def create_optimizer(
+            model, fp16=False, optimizer=None, **optimizer_params
+    ):
+        if optimizer is None:
+            return None
+
+        master_params = list(
+            filter(lambda p: p.requires_grad, model.parameters())
+        )
+        if fp16:
```

Comment:
let's create `catalyst.dl.fp16`'s function `def to_fp16` and move all from this if to the methos